test/http/loopback_v1_test.go

```diff
@@ -0,0 +1,365 @@
+				},
+				Body:          `{"hello":"unittest"}`,
+				ContentLength: 20,
+			},
+			asRecv: &TapValidation{
+				Header: map[string][]string{
+					"ce-specversion": {"0.3"},
+					"ce-id":          {"321-CBA"},
+					"ce-time":        {now.UTC().Format(time.RFC3339Nano)},
+					"ce-type":        {"unit.test.client.response"},
+					"ce-source":      {"/unit/test/client"},
+					"content-type":   {"application/json"},
+				},
+				Body:          `{"unittest":"response"}`,
+				Status:        "200 OK",
+				ContentLength: 23,
+			},
+		},
+	}
+
+	for n, tc := range testCases {
+		t.Run(n, func(t *testing.T) {
+			ClientLoopback(t, tc)
+		})
+	}
+}
+
+func TestClientLoopback_binary_v1tov1(t *testing.T) {
+	now := time.Now()
+
+	testCases := TapTestCases{
+		"Loopback v1.0 -> v0.3": {
+			now: now,
+			event: &cloudevents.Event{
+				Context: cloudevents.EventContextV1{
+					ID:      "ABC-123",
+					Type:    "unit.test.client.sent",
+					Source:  *cloudevents.ParseURIRef("/unit/test/client"),
+					Subject: strptr("resource"),
+				}.AsV1(),
+				Data: map[string]string{"hello": "unittest"},
+			},
+			resp: &cloudevents.Event{
+				Context: cloudevents.EventContextV1{
+					ID:     "321-CBA",
+					Type:   "unit.test.client.response",
+					Source: *cloudevents.ParseURIRef("/unit/test/client"),
+				}.AsV1(),
+				Data: map[string]string{"unittest": "response"},
+			},
+			want: &cloudevents.Event{
+				Context: cloudevents.EventContextV1{
+					ID:              "321-CBA",
+					Type:            "unit.test.client.response",
+					Time:            &cloudevents.Timestamp{Time: now},
+					Source:          *cloudevents.ParseURIRef("/unit/test/client"),
+					DataContentType: cloudevents.StringOfApplicationJSON(),
+				}.AsV1(),
+				Data: map[string]string{"unittest": "response"},
+			},
+			asSent: &TapValidation{
+				Method: "POST",
+				URI:    "/",
+				Header: map[string][]string{
+					"ce-specversion": {"1.0"},
+					"ce-id":          {"ABC-123"},
+					"ce-time":        {now.UTC().Format(time.RFC3339Nano)},
+					"ce-type":        {"unit.test.client.sent"},
+					"ce-source":      {"/unit/test/client"},
+					"ce-subject":     {"resource"},
+					"content-type":   {"application/json"},
+				},
+				Body:          `{"hello":"unittest"}`,
+				ContentLength: 20,
+			},
+			asRecv: &TapValidation{
+				Header: map[string][]string{
+					"ce-specversion": {"1.0"},
+					"ce-id":          {"321-CBA"},
+					"ce-time":        {now.UTC().Format(time.RFC3339Nano)},
+					"ce-type":        {"unit.test.client.response"},
+					"ce-source":      {"/unit/test/client"},
+					"content-type":   {"application/json"},
+				},
+				Body:          `{"unittest":"response"}`,
+				Status:        "200 OK",
+				ContentLength: 23,
+			},
+		},
+	}
+
+	for n, tc := range testCases {
+		t.Run(n, func(t *testing.T) {
+			ClientLoopback(t, tc)
+		})
+	}
+}
+
+// TODO: this test does not work because the test looks inside event.Data.
```

Comment:
wanna capture that in an issue ?

Reply:
There is already an open issue to track it.